### ISSUE TITLE
Not enough money for demo account

### ISSUE THREAD

**Issue by copypasteearth** [EA31337/EA31337#341]
Hi Im trying to setup EA with mt4, and every time it tries to trade there is this error
2022.07.26 19:01:04.033	EA31337-Lite-v2.010 EURUSD,M1: 2022.07.27 02:00:53: ERROR: Trade::OrderAdd:663: Cannot add order!; Code: 134, Msg: Not enough money.

Maybe its something im doing wrong, I log into my demo account and see this when I connect the expert
![bot](https://user-images.githubusercontent.com/6181334/181127368-ef6791d4-4e35-44a3-9229-40d756639576.PNG)

Thank you for your time

**Comment by kenorb** [comment]
Are you running with the default settings? It could only happen when you set lot size too large. Set 'Lot size' to '0' in EA's parameters, if it's already, otherwise set to minimal of 0.01 (it's possible auto lot calculate fails due to specific broker settings).

Can you also paste the initial screen before that right after start (which should include values with Market values).

**Comment by copypasteearth** [comment]
@kenorb I had the lot size set at 1000, I thought it was units, Im really new to this, so I set lot size to 0.01 and everything is good now.  It has traded a bunch of times but there are like 100s of these messages popping up, can you tell me what this means?
2022.07.27 05:10:00.339	EA31337-Lite-v2.010 EURUSD,M1: 2022.07.27 12:09:06: ERROR: Trade::OrderAdd:663: Cannot add order!; Code: 149, Msg: An attempt to open an order opposite to the existing one when hedging is disabled

**Comment by copypasteearth** [comment]
Im getting this also 
2022.07.27 05:03:03.472	EA31337-Lite-v2.010 EURUSD,M1: 2022.07.27 12:02:01: WARNING: Order::OrderModify:1086: Failed to modify order (#299436977/p:0/sl:1.01261/tp:0/code:136).; EURUSD,1.01424,1.0141,0,0,1e-05,0.0001,1e-05,1e-05,1,0,0,5,17,0,100000,0.01,100000,0.01,0,-0.55,0.06,0,0,0

**Comment by copypasteearth** [comment]
@kenorb How can I limit EA to one order at a time? I think that would solve the problem

**Comment by copypasteearth** [comment]
It is not being able to close later orders other than earlier orders and its getting stuck trying to close a more recent order when there are earlier orders

**Comment by kenorb** [comment]
- `1000` value for lot size is wrong, so there is some EA's miscalculation based on the broker's market values.
- error code 136 sounds like related to _Off quotes_ as per https://book.mql4.com/appendix/errors, sounds like another miscalculation or price changing too quickly

> An attempt to open an order opposite to the existing one when hedging is disabled

It seems your broker doesn't support hedging. It's not recommended to use brokers that don't support it.

For the workaround, you can change `Signal open filter` to `40` as per https://github.com/EA31337/EA31337/wiki/Input-parameters to activate 8th filter (do not open trades when there are opposite active trades).

**Comment by kenorb** [comment]
Please share with which broker you've got the issues, so I'll try to reproduce these.

**Comment by copypasteearth** [comment]
@kenorb I am using Oanda demo account

**Comment by copypasteearth** [comment]
@kenorb Im trying your workaround now, Ill let you know how it goes, it may be a US thing the mail says "filling order is against FIFO requirement". thats in the old case, ill let you know how the workaround works. Can you tell me a broker that this would definetly work with if it is a oanda thing, or one that has the hedging option?

**Comment by copypasteearth** [comment]
@kenorb yeah setting 40 did not work. It is still doing multiple orders. Is there a way i can make it only deal with one order at a time?

**Comment by kenorb** [comment]
> It is still doing multiple orders. Is there a way i can make it only deal with one order at a time?

That filter is per each strategy, try to set only one strategy (other set to None), then try SOF filter=40.

**Comment by kenorb** [comment]
> it may be a US thing the mail says "filling order is against FIFO requirement"

If that's a US thing, not sure if you can do much. In EU most of the broker can trade with hedging.

**Comment by copypasteearth** [comment]
@kenorb this is what I have set, I set the 40, and disabled the other strategies and it is still opening multiple orders
![Screen Shot 2022-07-27 at 8 04 55 PM](https://user-images.githubusercontent.com/6181334/181392920-2428fdbf-c8a6-477d-8688-61b56c8deebc.png)

**Comment by kenorb** [comment]
> opening multiple orders

The configuration looks fine. Opening multiple orders should be fine as long they're in the same direction. If you see in the different direction (buy+sell) at the same time (generating hedging error), then it's not good, so something needs fixing.

**Comment by copypasteearth** [comment]
It tries to modify the orders and I get the FIFO error again
